### DISCUSSION TITLE
Homematic (Bidcos-RF): HM-ES-TX-WM. Die Entität hat keine Zustandsklasse mehr..

### DISCUSSION THREAD

**Discussion by ThomasLoeffel** [SukramJ/aiohomematic#1746] [General]
### The problem

Seit dem neuesten Update (1.66.0) zeigt HM mir diese Fehlermeldung für GAS und Wasser an. Bei Strom funktioniert es wie gehabt. Der Dienst zeigt in den Statistiken aber alles wie gewünscht an. Er bietet zwei Reparaturen an. Indem er anbietet: Möchtest du die Langzeitstatistiken von sensor.hausanschluss_gas_gas_leistung dauerhaft aus deiner Datenbank löschen? Ich bin verwirrt!

Welche Infos braucht Ihr noch von mir?

### What version of HomematicIP (local) has the issue?

1.66.0

### What was the last working version of HomematicIP (local)?

1.65.0

### What type of installation are you running?

Home Assistant OS

### What type of installation are you running for your homematic backend?

RaspberryMatic Standalone

### Which version of your homematic backend are you running?

3.77.7.20240826

### What hardware are you running for your system?

Proxmox

### Which config details do you use

- [ ] TLS
- [X] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [X] Homematic IP
- [X] Homematic (Bidcos-RF)
- [X] Groups (Heating-Group)
- [X] BidCos-Wired (HM-Wired)

### Diagnostics information (no logs here)

[config_entry-homematicip_local-2557ef9295230914947af7c750f9e0c1.json](https://github.com/user-attachments/files/17261461/config_entry-homematicip_local-2557ef9295230914947af7c750f9e0c1.json)

### Additional information

Danke für Eure Unterstützung!
Gruß Thomas

**Comment by SukramJ** [comment]
Dann zeig doch mal wie die Entitäten unter Einstellungen/Entwicklerwerkzeuge aussehen.
An der Integration hat sich diesbezüglich nichts geändert, und aus deiner Beschreibung werde ich nicht schlau.

**Comment by SukramJ** [comment]
Seit Mai gibt es die [state_class](https://developers.home-assistant.io/docs/core/entity/sensor#long-term-statistics) (MEASUREMENT) nicht mehr bei GAS_POWER (sensor.hausanschluss_gas_gas_leistung), da eine Kombination, laut HA Definition, nur mit TOTAL oder TOTAL_INCREASING erlaubt ist. Neu mit HA 2024.10 ist, das die Fehler als "Repair" angezeigt werden.
Ist der Wert von` sensor.hausanschluss_gas_gas_leistung` stetig steigend (TOTAL*), oder wird die momentane Leistung (MEASUREMENT) angezeigt?

**Comment by PummelQualle** [comment]
Hallo, ich habe den selben Hinweis.

stetig steigend würde ich sagen:

![Bildschirmfoto_5-10-2024_9741_192 168 178 111](https://github.com/user-attachments/assets/549e777b-dc7a-49bb-9b87-f4eb57a67b75)

**Comment by SukramJ** [comment]
Das ist eine andere Entität!  Du zeigst GAS_ENERGY_COUNTER(Energiezähler), aber es geht um GAS_POWER(Leistung).

**Comment by PummelQualle** [comment]
Ok dann der Reihe nach.

mit dem neuesten HA Update habe ich folgende Meldung erhalten:
![Bildschirmfoto_5-10-2024_91729_192 168 178 111](https://github.com/user-attachments/assets/a16acc4f-3869-4219-8638-44503b7daad6)
 ich habe nur 2 Entitäten dafür:
![Bildschirmfoto_5-10-2024_91910_192 168 178 111](https://github.com/user-attachments/assets/92c620a6-bfec-4dbe-ad43-900b8412f63e)
und der gepostete ist bei mir im Energiezähler hinterlegt:

![Bildschirmfoto_5-10-2024_91943_192 168 178 111](https://github.com/user-attachments/assets/742e8d18-7fda-489d-aaf1-88bd790815f8)

**Comment by SukramJ** [comment]
In deiner Fehlermeldung steht klar ...gas_leistung. Gas-Energiezähler und die Einbindung ins Energie-Dashboard sind irrelevant.

**Comment by PummelQualle** [comment]
Sorry Screenshot Gas Leistung habe ich vergessen.

der sieht bei mir aber merkwürdig aus:

![Bildschirmfoto_5-10-2024_94527_192 168 178 111](https://github.com/user-attachments/assets/b0e85740-decf-4334-bf94-859e2fa6c059)
![Bildschirmfoto_5-10-2024_94527_192 168 178 111](https://github.com/user-attachments/assets/725ae25f-471c-4d09-b1a4-af522aabaa95)

**Comment by PummelQualle** [comment]
![Bildschirmfoto_5-10-2024_9454_192 168 178 111](https://github.com/user-attachments/assets/9717c018-075c-4cb6-82db-8743726f1ea8)

**Comment by SukramJ** [comment]
Der Grund warum im Mai die `state_class` entfernt wurde:
Der Hersteller liefert als Einheit Kubikmeter, und schreibt das so auch in seiner Doku.
Mit Kubikmeter ist in HA aber nur die TOTAL oder TOTAL_INCREASING state_class erlaubt.
Also wurde die `state_class` aufgrund einer HA Fehlermeldung entfernt.
Mit HA 2024.10 hat HA die Statistik-Fehlermeldungen umgestellt.
Der Fehler sollte seit Mai/Juni auch unter Entwicklerwerkzeuge/Statistik zu sehen gewesen sein.

Ich werde in der Integration nichts ändern, da ich ungern etwas vom Hersteller geliefertes verbiege.

Ihr könnt euch die Entität aber in HA konfigurieren (customize.yaml):

```
sensor.hausanschluss_gas_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement
```
Ich hab das mangels Sensor nicht getestet.

Es dauert einige Stunden bis die Statistiken neu berechnet werden, also warten!

**Reply by ThomasLoeffel** [reply]
Danke für den Tip! Frage: wo steht die "customize.yaml" und wie kann ich diese öffnen? Ich bin da nicht so der Profi :-).

Habe Dir einen Studio Code Server Ausdruck beigefügt.

Gruß Thomas


![20101005_HA_Studio_!](https://github.com/user-attachments/assets/371a49f3-eff5-411c-a6f8-d3d2241b7c70)

**Reply by ThomasLoeffel** [reply]
Sorry, habe es in der Doku gefunden. Teste es jetzt.

**Reply by ThomasLoeffel** [reply]
Hallo Zusammen,

mal eine Frage. Ich benutze den HM-ES-TX-WM für Strom, Gas und Wasser. Wobei bei Wasser die selben Einstellungen wie bei Gas auf der Homematic (nur die Zählerkonstante ist angepasst) hinterlegt ist. Ich habe nun wie besprochen die beiden Sensoren angepasst und schaue ob die Warnungen wiederkehren. Was muss ich beim Wassersensor einstellen damit er in den Energie unter Wasser angezeigt werden kann. Die Werte liefert er ja.

Gruß und Danke Thomas

# Anpassung für Gas-Sensor / Wasser - Sensor HMIP
sensor.hausanschluss_gas_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement

sensor.hausanschluss_wasser_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement

![20101005_HA_Studio_Wasser](https://github.com/user-attachments/assets/09e9b736-bb85-4eab-aba3-e171b9d8904c)

**Reply by SukramJ** [reply]
Es fehlt die device_class:

```
sensor.hausanschluss_gas_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement
  device_class: volume_flow_rate

sensor.hausanschluss_wasser_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement
  device_class: water
```

**Reply by ThomasLoeffel** [reply]
Danke für den Tip!

Dieses geht allerdings nicht (device_class: volume_flow_rate)! Richtiger wäre vielleicht "gas", dieses habe ich eingestellt.
Aber im Energie Bereich findet er trotz allem den  HM-ES-TX-WM Wasser nicht. Noch eine Idee was ich falsch mache?

Gruß Thomas

**Reply by SukramJ** [reply]
> Noch eine Idee was ich falsch mache?

Gas ist nicht richtig. [Da HA bei gas state_class MEASURE nicht aktzeptiert.](https://github.com/home-assistant/core/blob/4693f436a58e9038dda6ca9cb6803fa5871ddef3/homeassistant/components/sensor/const.py#L618) , und für [Gas nur m3 zugelassen ist](https://github.com/home-assistant/core/blob/4693f436a58e9038dda6ca9cb6803fa5871ddef3/homeassistant/components/sensor/const.py#L203-L209).

Deswegen machen wir das hier doch!

Für das Energy Dashboard solltest Du nur die `ENERGY_COUNTER` Entitäten verwenden. Hier muss ggf. die `device_class` bei Wasser auf `water` geändert werden.

Für `sensor.hausanschluss_wasser_gas_leistung` muss die device_class auch `volume_flow_rate` sein.

**Reply by ThomasLoeffel** [reply]
Danke für Deine Info.

Anbei wo ich was eingestellt habe. Hiermit funktioniert es nicht. Auch der Eintrag: volume_flow_rate verursacht Fehler.
Kannst Du/Ihr mir helfen wo ich was genau ändern sollte.

PS. Der Wassersensor ist im Energie unter GAS auswählbar, nicht im Wasser.
Gruß Thomas

So sind meine Einträge:
![20241006_Sensor Wasser_customize](https://github.com/user-attachments/assets/31df2045-d5b7-41c0-8c82-1ca41c5b9d22)
![20241006_Sensor Wasser](https://github.com/user-attachments/assets/c0dd3eaa-e5c7-4cdd-9304-fdef2cead3ae)
![20241006_Sensor Wasser_detail](https://github.com/user-attachments/assets/5c467aea-9fae-406d-80ad-4771ff6b5402)


Hier der Versuch "volume_flow_rate" anzuwenden:
![20241006_Sensor Wasser_customize_Fehler](https://github.com/user-attachments/assets/afb5f31f-cb05-4c47-90ee-1e3d19f0e9e8)

**Reply by SukramJ** [reply]
Die Fehler im VSCode sind egal. Wichtiger ist, das im HALog keine Fehler sind.

Wie oben geschrieben solltest du die jeweils anderen  Entitäten ( Energiezähler/ Wasser-Zähler) verwenden, und die entsprechenden Anpassungen (siehe oben) machen. Dann ist aber water als device_class richtig.

Bei Statistiken musst du immer warten(ggf. Stunden), da ein zyklischer Job läuft.
 Unter Entwicklertools/ Statistiken musst du ggf. noch die Fehler korrigieren.

Ein Blick in die HA Doku schadet auch nicht.

**Reply by ThomasLoeffel** [reply]
Ich danke Dir wirklich für Deine Hilfe. Leider ist mein Niveau gegenüber Deinem wohl erheblich niedriger. Ich verstehe nicht was Du genau meinst :-(. Ich lese die Doku aber mir erschließt sich nicht alles.

Gas funktioniert es geht mir nur um den Wasserzähler den ich in der Energieübersicht auswählen möchte (derzeit geht es nicht).
Ich habe dieses in der customize.yaml eingegeben.

Was sollte ich hier genau ändern?
.....................................................................................................
sensor.hausanschluss_wasser_gas_leistung:
  unit_of_measurement: 'm³/h'
  state_class: measurement
  device_class: water
....................................................................................................

Was soll ich wo bei den Entitäten ändern. Ich danke Dir für den erneuten Hinweis, aber es erschließt sich mir noch noch nicht was ich genau wo in was ändern soll.

Sorry und vielen Dank für die Geduld.
Gruß Thomas

**Reply by ThomasLoeffel** [reply]
So habe es hinbekommen.

Habe dem Artikel von JM gelesen und verstanden. Ich musste den Energiezähler anpassen und auch in den Statistiken die Fehler beheben.

sensor.hausanschluss_wasser_gas_energiezahler:
  state_class: total_increasing 
  unit_of_measurement: 'm³' 
  device_class: water


Das System rechnet wohl automatisch von m3 auf Liter um.

Danke für die Hilfe, Fall geschlossen :-).
Gruß Thomas

![20241007_Sensor_Energie](https://github.com/user-attachments/assets/704b3bea-4133-4d13-8b71-68601e70d68f)
![20241007_Sensor_customize](https://github.com/user-attachments/assets/984ee090-0cc9-4cb0-89c0-730c54e76533)
![20241007_JM_Rueckfrage_1](https://github.com/user-attachments/assets/269632b4-3758-439b-8a53-785718647587)
![20241007_JM_Antwort](https://github.com/user-attachments/assets/980583e0-f682-44d1-9f3b-3a880e5db8ae)

**Comment by Baxxy13** [comment]
Es muss doch nur bei dem HM-ES-TX-WM der am Wasserzähler hängt für `gas_energiezahler` die Device-Class (per Customize) zu `water` geändert werden.
Beispiel:
```
 customize:
   sensor.hm_es_tx_wm_peq0543441_gas_energiezahler:
     device_class: water
```

Mehr nicht.

**Reply by ThomasLoeffel** [reply]
Wohl wahr.... wenn man das Gesamtsystem nicht richtig versteht dauert es bis der Groschen fällt. Jetzt geht es ja und habe viel gelernt auf dem neuen Gebiet des HA.

Danke für die Geduld.
Thomas
